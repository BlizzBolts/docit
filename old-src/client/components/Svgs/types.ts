import type { SVGAttributes } from "react";
import type React from "react";

export interface SvgProps extends SVGAttributes<SVGElement> {
  className?: string;
  style?: React.CSSProperties;
}
