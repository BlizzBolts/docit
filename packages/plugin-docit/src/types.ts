export interface DocitPluginOption {
  root: string;
}
