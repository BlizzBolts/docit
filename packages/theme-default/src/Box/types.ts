import React from "react";

export interface BoxProps {
  color?: string;
  style: React.CSSProperties;
}
