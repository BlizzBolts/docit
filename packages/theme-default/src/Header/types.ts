import type { BaseProps } from "@/types";

export interface HeaderProps extends BaseProps {
  title: string;
}
