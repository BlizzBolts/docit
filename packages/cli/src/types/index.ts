export type ConfigFileLocation = string;

export interface CLIOptions {
  config?: ConfigFileLocation;
}
