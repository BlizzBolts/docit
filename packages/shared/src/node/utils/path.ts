import path from "node:path";
import url from "node:url";
/**
 * isomorphic __dirname getter
 *
 * @param {import.meta.url} importMetaUrl
 * @returns
 */
export const getDirname = (importMetaUrl: string) => {
  return path.dirname(url.fileURLToPath(importMetaUrl));
};
