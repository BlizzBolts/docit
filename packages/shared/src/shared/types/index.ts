export * from "./docit-config";
export * from "./scaffold-options";
export * from "./server-config";
export * from "./site-config";
export * from "./theme-config";
