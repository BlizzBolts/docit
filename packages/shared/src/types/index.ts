import type { PackageJson } from "type-fest";

export { PackageJson };
