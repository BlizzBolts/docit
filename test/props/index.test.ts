import { expect, describe, it } from "vitest";
import path from "path";
import {
  parser,
  parseApi,
  tableMaker,
  parseTreeToMdx,
} from "../../src/node/utils/index.js";
import { ResolvedComponentProps } from "../../src/node/types.js";

const componentPath = path.resolve(__dirname, "./Button.tsx");

describe("Props And Api Parsing", () => {
  it("matctes raw parsed result", () => {
    const parsed = parser.parse(componentPath);
    expect(parsed).toMatchSnapshot();
  });

  it("parses result", () => {
    const parsed = parseApi(componentPath);
    expect(parsed).toMatchSnapshot();
  });

  it("generates correct table", () => {
    const { t, r, c } = tableMaker;

    const heading = r([
      c("属性"),
      c("类型"),
      c("默认值"),
      c("必填"),
      c("描述"),
    ]);

    const parsed: ResolvedComponentProps[] = parseApi(componentPath);

    const tables = parsed.map((o) => {
      console.log(o.componentName, o.filePath);
      return t([
        heading,
        ...o.props?.map((p) => {
          return r([
            c(p.name || "-"),
            c(p.type || "-"),
            c(p.defaultValue || "-"),
            c(p.isRequired ? "是" : "否"),
            c(p.description || "-"),
          ]);
        }),
      ]);
    });

    expect(
      parseTreeToMdx({
        type: "root",
        children: tables,
      })
    ).toMatchSnapshot();
  });
});
