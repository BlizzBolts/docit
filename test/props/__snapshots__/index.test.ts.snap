// Vitest Snapshot v1

exports[`Props And Api Parsing > generates correct table 1`] = `
"| 属性 | 类型 | 默认值 | 必填 | 描述 |
| -- | -- | --- | -- | -- |

| 属性      | 类型                | 默认值          | 必填 | 描述 |
| ------- | ----------------- | ------------ | -- | -- |
| color   | string            | grene        | 否  | -  |
| length  | Length            | Length.Large | 否  | -  |
| variant | \\"A\\" \\\\| \\"B\\" \\\\| \\"C\\" | -            | 是  | -  |
"
`;

exports[`Props And Api Parsing > matctes raw parsed result 1`] = `
[
  {
    "description": "",
    "displayName": "StyledButton",
    "filePath": "/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/test/props/Button.tsx",
    "methods": [],
    "props": {},
    "tags": {},
  },
  {
    "description": "",
    "displayName": "Button",
    "filePath": "/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/test/props/Button.tsx",
    "methods": [],
    "props": {
      "color": {
        "declarations": [
          {
            "fileName": "docgen/test/props/Button.tsx",
            "name": "ButtonProps",
          },
        ],
        "defaultValue": {
          "value": "grene",
        },
        "description": "一种颜色",
        "name": "color",
        "parent": {
          "fileName": "docgen/test/props/Button.tsx",
          "name": "ButtonProps",
        },
        "required": false,
        "type": {
          "name": "string",
        },
      },
      "length": {
        "declarations": [
          {
            "fileName": "docgen/test/props/Button.tsx",
            "name": "ButtonProps",
          },
        ],
        "defaultValue": {
          "value": "Length.Large",
        },
        "description": "",
        "name": "length",
        "parent": {
          "fileName": "docgen/test/props/Button.tsx",
          "name": "ButtonProps",
        },
        "required": false,
        "type": {
          "name": "enum",
          "raw": "Length",
          "value": [
            {
              "description": "大",
              "fullComment": "大",
              "tags": {},
              "value": "5",
            },
            {
              "description": "中",
              "fullComment": "中",
              "tags": {},
              "value": "3",
            },
            {
              "description": "小",
              "fullComment": "小",
              "tags": {},
              "value": "1",
            },
          ],
        },
      },
      "variant": {
        "declarations": [
          {
            "fileName": "docgen/test/props/Button.tsx",
            "name": "ButtonProps",
          },
        ],
        "defaultValue": null,
        "description": "",
        "name": "variant",
        "parent": {
          "fileName": "docgen/test/props/Button.tsx",
          "name": "ButtonProps",
        },
        "required": true,
        "type": {
          "name": "enum",
          "raw": "\\"A\\" | \\"B\\" | \\"C\\"",
          "value": [
            {
              "value": "\\"A\\"",
            },
            {
              "value": "\\"B\\"",
            },
            {
              "value": "\\"C\\"",
            },
          ],
        },
      },
    },
    "tags": {},
  },
]
`;

exports[`Props And Api Parsing > parses result 1`] = `
[
  {
    "componentName": "StyledButton",
    "filePath": "/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/test/props/Button.tsx",
    "props": [],
  },
  {
    "componentName": "Button",
    "filePath": "/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/test/props/Button.tsx",
    "props": [
      {
        "defaultValue": "grene",
        "enums": undefined,
        "isEnum": false,
        "isRequired": false,
        "name": "color",
        "type": "string",
      },
      {
        "defaultValue": "Length.Large",
        "enums": [
          {
            "description": "大",
            "value": "5",
          },
          {
            "description": "中",
            "value": "3",
          },
          {
            "description": "小",
            "value": "1",
          },
        ],
        "isEnum": true,
        "isRequired": false,
        "name": "length",
        "type": "Length",
      },
      {
        "defaultValue": undefined,
        "enums": [
          {
            "description": undefined,
            "value": "\\"A\\"",
          },
          {
            "description": undefined,
            "value": "\\"B\\"",
          },
          {
            "description": undefined,
            "value": "\\"C\\"",
          },
        ],
        "isEnum": true,
        "isRequired": true,
        "name": "variant",
        "type": "\\"A\\" | \\"B\\" | \\"C\\"",
      },
    ],
  },
]
`;
